enhancements/two-nodes-openshift/2no.md

```diff
@@ -0,0 +1,516 @@
+
+Split-brain - a scenario where a set of peers are separated into groups smaller than the quorum threshold AND peers decide to host services already running by other groups.  Typically results in data loss or corruption.   
+
+MCO - Machine Config Operator. This operator manages updates to node's systemd, cri-o/kubelet, kernel, NetworkManager, etc., and can write custom files to it, configurable by MachineConfig custom resources.
+
+ABI - Agent-Based Installer.
+
+## Summary
+
+Leverage traditional high-availability concepts and technologies to provide a container management solution suitable for customers with numerous geographically dispersed locations that has a minimal footprint but remains resilient to single node-level failures.
+
+## Motivation
+
+Customers with hundreds, or even tens-of-thousands, of geographically dispersed locations are asking for a container management solution that retains some level of resilience to node level failures, but does not come with a traditional three-node footprint and/or price tag.
+
+The need for some level of fault tolerance prevents the applicability of Single Node OpenShift (SNO), and a converged 3-node cluster is cost prohibitive at the scale of retail and telcos - even when the third node is a "cheap" one that doesn't run workloads.
+
+The benefits of the cloud-native approach to developing and deploying applications are increasingly being adopted in edge computing.
+This requires our solution to provide a management experience consistent with "normal" OpenShift deployments, and be compatible with the full ecosystem of Red Hat and partner workloads designed for OpenShift.
+
+### User Stories
+
+* As a large enterprise with multiple remote sites, I want a cost-effective OpenShift cluster solution so that I can manage containers without the overhead of a third node.
+* As a support engineer, I want a safe and automated method for handling the failure of a single node so that the downtime of the control-plane is minimized.
+
+### Goals
+
+* Provide a two-node control-plane for physical hardware that is resilient to a node-level failure for either node
+* Provide a transparent installation experience that starts with exactly 2 blank physical nodes, and ends with a fault-tolerant two node cluster
+* Prevent both data corruption and divergent datasets in etcd
+* Maintain the existing level of availability. Eg. by avoiding fencing loops, wherein each node powers cycles it's peer after booting, reducing the cluster's availability.
+* Recover the API server in less than 120s, as measured from the surviving node's detection of a failure
+* Minimize any differences to the primary OpenShift platforms
+* Avoid any decisions that would prevent future implementation and support for upgrade/downgrade paths between two-node and traditional architectures 
+* Provide an OpenShift cluster experience that is identical to that of a 3-node hyperconverged cluster, but with 2 nodes
```

Comment:
Just trying to indicate that there's nothing else, no dedicated workers. Anything the cluster needs to do is done on and by those two nodes.

Is there a better term to use here?

Reply:
Compact cluster = combined control plane/worker nodes with no separate workers.
[Hyperconverged](https://en.wikipedia.org/wiki/Hyper-converged_infrastructure) usually refers to using on-cluster-node shared storage like Ceph instead of a SAN.

Reply:
There won't be a SAN (storage will be discussed in a followup enhancement), so I think hyperconverged is the correct term to use here

Reply:
If [ODF](https://www.redhat.com/en/technologies/cloud-computing/openshift-data-foundation) is part of it then it is hyperconverged.